addons/bus/static/src/workers/websocket_worker.js

```diff
@@ -395,8 +402,13 @@ export class WebsocketWorker {
     }
 
     _logDebug(title, ...args) {
-        if (this.loggingEnabled) {
-            logger.log({ dt: new Date().toISOString(), event: title, args, worker: UUID });
+        try {
+            if (this.loggingEnabled) {
+                logger.log({ dt: new Date().toISOString(), event: title, args, worker: UUID });
+            }
+        } catch (e) {
+            console.error("Logger is not available, falling back to console.");
+            console.error(e);
```

Comment:
What error is this supposed to catch, and what's the point of using console if we won't receive it's output anyway?

Also shouldn't the `try` be inside the `if` rather than outside?

Reply:
We are investigating problems with self im_status being offline, for which we still haven't found a way to reproduce the issue.

When the issue happens, we saw that a debounced function is not called when we expect it should. At a glance of code, one theory for bug is silent crash in debounced function. This PR puts code to catch such errors so that when the problem happens again we can see whether the cause is crash and fix it thanks to log.

`console.error()` there is a last resort, e.g. when the logger itself causes the crash. The `try` above the `if()` is to give more guarantee to not trigger infinite loop of crash.